liferay-echo/liferay-portal
# [TECHNICAL SUPPORT] LPS-203529 Don't allow the removal of Collections linked to Pages.

### 🐛  About the Issue
**Deleting** a Collection does **not** delete the **Page** to which it is connected. After deleting the Collection, the previously mentioned Page appears **broken**, and the following error is logged:
`com.liferay.asset.list.exception.NoSuchEntryException: No AssetListEntry exists with the primary key 31770`

### 💡 Desired Solution
**Don't allow** the removal of Collections linked to Pages.

### 📖 PR Content
- **LPS-203529**
  - Imported the `RequiredAssetListEntryException` in the asset-list-web `init.jsp`. 
  - Added a **custom error tag** to the asset-list-web `view.jsp`.
  - Created the `AssetListEntryServiceImpl._checkCompanyAssetListEntryUsages` method. 
  - Extended the logic of the `AssetListEntryServiceImpl.deleteAssetListEntries` and `AssetListEntryServiceImpl.deleteAssetListEntry` methods with the usage of the new method.
  - Extended the `Language.properties` with the `this-asset-cannot-be-deleted-because-it-is-in-use` error message.
  - Created the `RequiredAssetListEntryException` class.

### 🧪 How To Test
1. Go to `Site Builder` > `Collections`.
2. Create a **new Dynamic Collection** restricted to `Web Content Article` > `Basic Web Contents`.
3. Go to `Site Builder` > `Pages` > Add a **new Collection Page** based on the one created in the previous step > **Publish**.
4. Return to `Site Builder` > `Collections`.
5. **Delete** the collection with the ⋮ menu.
6. Return to the collection page **and observe** it.

### ✅ Expected Result
When **attempting** to delete a **Collection** _(or any Asset List Entry)_ that **has usages**, an error message should appear with the following: "`This asset cannot be deleted because it is in use.`"
